offchain/types/src/foldables.rs

```diff
@@ -82,6 +87,7 @@ impl Foldable for InputBox {
         env: &StateFoldEnvironment<M, Self::UserData>,
         access: Arc<FoldMiddleware<M>>,
     ) -> Result<Self, Self::Error> {
+        let dapp_address = Arc::clone(&previous_state.dapp_address);
```

Comment:
I think you could also check the bloom filter for `dapp_address`.

Reply:
@GCdePaula I added `fold_utils::contains_address(&block.logs_bloom, &dapp_address)` to the filter, but the dispatcher stopped reading inputs. Do you have a more specific suggestion?

Reply:
Never mind; it worked with `fold_utils::contains_topic(&block.logs_bloom, &*dapp_address)`